02-intermediate-ruby/api-post.md

```diff
@@ -0,0 +1,282 @@
+# Making Post Requests
+
+In our past exercises we _consumed_ information from an API.  We looked up details about particular resources and used them in our Ruby applications.  However we do not yet have the ability to _create_ any information on the external API.  In this lesson we will introduce the concept of a `POST` request which can save information to the API.  We will do so using the [Slack API](https://api.slack.com/) to create our own command-line slack client.
+
+## Learning Goals
+
+By the end of this lesson you should be able to:
+
+- Identify the differences between a `GET` and `POST` request.
+- Use Postman and HTTParty to make a `POST` request to save data to an api.
+- Write tests to verify the results of a `POST` request using the VCR gem.
+
+## Overview of Post Requests
+
+With `GET` requests you request information from the API.  The API then sends information back in the body of the response.  However with a `GET` request, you only need specify the method and URL of the request.  Any other information is embedded in the URL as _Query Parameters_.
+
+With a `POST` request you are asking the API to accept some data and create a resource.  For example, a blogging API like the [Wordpress API](https://developer.wordpress.org/rest-api/reference/posts/#create-a-post) takes a date, author, title and body when it creates a new blog entry.  Because you are sending additional information with all `POST` requests, these requests have an additional _body_ section to contain the information being sent.
+
+Below is a diagram of the structure of a `POST` request.
+
+![structure of an http request](images/http-request.png)
```

Comment:
This diagram feels a little misleading to me because the contents of the request's body has a thing named `response`. Could we change `response` to something else, like the Wordpress example you mentioned above's `author`, `date`, `title`, `body`?

Reply:
updated